dart/lib/src/sentry_client.dart

```diff
@@ -22,11 +23,14 @@ class SentryClient {
 
   final Random _random;
 
+  final SentryExceptionFactory _exceptionFactory;
+
   static final _sentryId = Future.value(SentryId.empty());
 
   /// Instantiates a client using [SentryOptions]
   SentryClient._(this._options)
-      : _random = _options.sampleRate == null ? null : Random();
+      : _exceptionFactory = SentryExceptionFactory(options: _options),
```

Comment:
how would you inject a `SentryExceptionFactory` for testing reasons?

Reply:
can we consider an optional ctor param with a default value?

Reply:
✅ 
```dart
SentryClient._(this._options, {SentryExceptionFactory exceptionFactory})
      : _exceptionFactory =
            exceptionFactory ?? SentryExceptionFactory(options: _options),
        _random = _options.sampleRate == null ? null : Random();
```